### ISSUE TITLE
[Bug]: Storybook 7 does not like "import type" in web component code from StencilJS

### ISSUE THREAD

**Issue by Kwooda** [storybookjs/storybook#21664]
### Describe the bug

I installed Storybook v7 RC4 to explore converting our stories and cannot get it to run. We build Web components using StencilJS, and the following code is included in a generated file from StencilJS:

```js
import { createReactComponent } from './react-component-lib';
import type { JSX } from '@our/library';

```

Where @our/library refers to a module that defines our components.
Storybook will launch, and I see a component tree in the left nav, but there is some step that builds within the browser, and a the following error occurs:

```
ModuleBuildError: Module build failed (from ./node_modules/@storybook/builder-webpack5/node_modules/babel-loader/lib/index.js):
SyntaxError: /Users/xxxxxx/projects/designsystem/kds-react/src/generated/framework-int/react/src/components.ts: Unexpected token, expected "from" (6:12)

  4 | import { createReactComponent } from './react-component-lib';
  5 |
> 6 | import type { JSX } from '@our/library';
    |             ^
  7 |
```

Either, we need to know what needs to be configured for it to parse import type properly, or if it's a bug that needs to be fixed in Storybook.

Another error is thrown in another generated file from StencilJS:

```
ModuleBuildError: Module build failed (from ./node_modules/@storybook/builder-webpack5/node_modules/babel-loader/lib/index.js):
SyntaxError: /Users/tkmafa3/projects/designsystem/kds-react/src/utils/core-init.ts: Missing semicolon. (4:4)

  2 | import { HTMLAttributes } from 'react'
  3 |
> 4 | type StencilToReact<T> = {
    |     ^
  5 |     [P in keyof T]?: T[P] & Omit<HTMLAttributes<Element>, 'className'> & {
  6 |     class?: string;
  7 |     };
```

This seems a bit strict to flag a missing semicolon as an error that aborts the build, especially in a generated file that we have no control over. Is there a way to relax the restriction and reduce it to a warning?

### To Reproduce

The last time I posted a repo for debugging, the security team jumped on me, so I'm not sure I can even post it, as I can't share our component library. I also ran npx sb@next sandbox locally to try to recreate the issue, and I'm not entirely sure how to replicate it there, as it asked for a template, and I have no idea which one to use. I picked what I thought was the closest, but since I know little about Storybook 7, I don't know the implications. It successfully launches Storybook, but none of my components are in there. I migrated one over, but it's not even rendering it, so I suspect there's something I need to add for it to recognize the component tags, which is also presumably the piece that fails with the issue I'm trying to reproduce. I'm trying my best to navigate this landscape; gone are the days when computers were simple. I'll update this report if there's any possible way I can make a repo available.

### System

```shell
Environment Info:

  System:
    OS: macOS 11.6
    CPU: (12) x64 Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz
  Binaries:
    Node: 16.15.0 - ~/.nvm/versions/node/v16.15.0/bin/node
    Yarn: 1.22.19 - ~/.nvm/versions/node/v16.15.0/bin/yarn
    npm: 6.14.18 - ~/projects/designsystem/kds-react/node_modules/.bin/npm
  Browsers:
    Chrome: 111.0.5563.64
    Safari: 15.0
  npmPackages:
    @storybook/addon-essentials: ^7.0.0-rc.4 => 7.0.0-rc.4 
    @storybook/addon-interactions: ^7.0.0-rc.4 => 7.0.0-rc.4 
    @storybook/addon-links: ^7.0.0-rc.4 => 7.0.0-rc.4 
    @storybook/blocks: ^7.0.0-rc.4 => 7.0.0-rc.4 
    @storybook/react: ^7.0.0-rc.4 => 7.0.0-rc.4 
    @storybook/react-webpack5: ^7.0.0-rc.4 => 7.0.0-rc.4 
    @storybook/testing-library: ^0.0.14-next.1 => 0.0.14-next.1
```

### Additional context

In summary, I'm not able to get a component built in StencilJS to render in Storybook 7.

**Comment by shilman** [comment]
Do you have a babel config? Storybook 7 now requires one for TS support:

https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#babel-mode-v7-exclusively

**Comment by Kwooda** [comment]
A babel config was generated by the migration script. Is there something I need to set in there?

**Comment by Flcwl** [comment]
why need it?

**Comment by Flcwl** [comment]
I have same error in monorepo project.

**Comment by aakarshitChoudhary** [comment]
I'm facing similar error with storybook
```
npm storybook Version 
 "@storybook/addon-essentials": "^7.4.1",
    "@storybook/addon-interactions": "^7.4.1",
    "@storybook/addon-links": "^7.4.1",
    "@storybook/addon-onboarding": "^1.0.8",
    "@storybook/addon-styling-webpack": "^1.0.0",
    "@storybook/blocks": "^7.4.1",
    "@storybook/react": "^7.4.1",
    "@storybook/react-webpack5": "^7.4.1",
    "@storybook/testing-library": "^0.2.0",
```
    
    Please help
